### ISSUE TITLE
Clear method throws error

### ISSUE THREAD

**Issue by aduth** [planttheidea/moize#52]
Trying to follow example from documentation results in error when testing:

```
$ npm show moize version
3.1.0
$ node
> const moize = require( 'moize' );
undefined
> memoized = moize((item) => {
...   return item;
... });
{ [Function: moizedFunction]
  cache: Cache { lastItem: {}, list: [], size: 0 },
  displayName: 'moize(function)',
  isMoized: true,
  options: 
   { equals: null,
     isPromise: false,
     isReact: false,
     maxAge: Infinity,
     maxArgs: Infinity,
     maxSize: Infinity,
     promiseLibrary: [Function: Promise],
     serialize: false,
     serializeFunctions: false,
     serializer: null },
  originalFunction: [Function],
  add: [Function],
  clear: [Function: clear],
  has: [Function],
  keys: [Function],
  remove: [Function],
  values: [Function] }
> memoized.clear();
TypeError: Cannot set property 'length' of undefined
    at Function.clear (/Users/andrew/Documents/Code/rememo/node_modules/moize/lib/Cache.js:65:22)
    at repl:1:10
    at ContextifyScript.Script.runInThisContext (vm.js:23:33)
    at REPLServer.defaultEval (repl.js:339:29)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
    at REPLServer.onLine (repl.js:536:10)
    at emitOne (events.js:101:20)
    at REPLServer.emit (events.js:191:7)
    at REPLServer.Interface._onLine (readline.js:241:10)
```

RequireBin demo: http://requirebin.com/?gist=f1c89ea433891eff414ffe8790b79d02

**Comment by planttheidea** [comment]
Ah interesting ... I tried to remove what I thought was an unneeded wrapper around the convenience `clear` method pointing to `cache.clear`, but I guess it is needed in certain scenarios. I'll release a fix for this shortly.

**Comment by planttheidea** [comment]
Alright, this should be fixed in `3.1.2` ... let me know if not and I'll reopen the issue.

**Comment by aduth** [comment]
Thanks for the quick fix!
